pkg/apis/pipeline/images.go

```diff
@@ -40,3 +42,25 @@ type Images struct {
 	// ImageDigestExporterImage is the container image containing our image digest exporter binary.
 	ImageDigestExporterImage string
 }
+
+// Validate returns an error if any image is not set.
+func (i Images) Validate() error {
```

Comment:
This will only show the first image that is "unset". Couldn't we add in the error *all* the images that are unset instead ?

Reply:
Good call, done. `Validate` is a bit uglier now, but the map approach didn't support multiple `""` keys.